### PR TITLE
new icon: vault (original, original-wordmark, plain-wordmark)

### LINKED ISSUE #1206: [ICON REQUEST]: HashiCorp Vault
### I have searched through the issues and didn't find my problem.

- [X] Confirm

### About the icon

Popular open source secrets engine. Frequently used alongside other HashiCorp products already included

### Links and sources

HashiCorp Brand Guidelines: https://www.hashicorp.com/brand#vault
Logo Bundle: https://drive.google.com/uc?export=download&id=1MWU8ODoxNFebk_MXlNlUWs98lqxqEqtN

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by luizcarloscf**
## Double check these details before you open a PR

- [x] PR does not match another non-stale PR currently opened
- [x] PR name matches the format *new icon: <i>Icon name</i> (<i>versions separated by comma</i>)*. More details [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#overview)
- [x] PR's base is the `develop` branch.
- [x] Your icons are inside a folder as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#organizational-guidelines)
- [x] SVG matches the standards laid out [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#svgStandards)
- [x] A new object is added in the `devicon.json` file as seen [here](https://github.com/devicons/devicon/blob/develop/CONTRIBUTING.md#-updating-the-deviconjson-)

**This PR closes #1206**

## Link to prove your SVG is correct and up-to-date.

https://www.hashicorp.com/brand#vault
https://www.vaultproject.io/

### CODE REVIEW

File: devicon.json [hunk 1]
```diff
@@ -9097,6 +9097,33 @@
         "color": "#a56de2",
         "aliases": []
     },
+    {
+        "name": "vault",
+        "altnames": [],
```
Comment (Snailedlt) [inline devicon.json:9102]:
```suggestion
        "altnames": ["HashiCorp Vault"],
```

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Just an alias I want to get in. Once that's added I'll approve ✔️

**Review by Snailedlt — CHANGES_REQUESTED** [review]
Same here as with #1604 :)

File: devicon.json [hunk 2]
```diff
@@ -9097,6 +9097,33 @@
         "color": "#a56de2",
         "aliases": []
     },
+    {
+        "name": "vault",
+        "altnames": ["HashiCorp Vault"],
```
Comment (lunatic-fox) [inline devicon.json:9102]:
```suggestion
        "altnames": [
            "HashiCorp Vault"
        ],
```

**Review by lunatic-fox — APPROVED** [review]
Approved! ✔
Thank you for this great contribution! 🚀

**Review by Snailedlt — APPROVED** [review]
Excellent PR as always! 🎉 
Approved! ✔️

**Pull request merged**